### PR TITLE
feat(docs): document `servicenow` package

### PR DESCRIPTION
**Pull request by pierwill**

### CODE REVIEW

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 1]
```diff
@@ -1,61 +1,77 @@
...
+//
+// ServiceNow Event API fields are described in
+// [ServiceNow Create Event documentation](https://docs.servicenow.com/bundle/paris-it-operations-management/page/product/event-management/task/t_EMCreateEventManually.html).
+//
+// - url: (Required) ServiceNow web service URL.
+// - username: (Required) ServiceNow username to use for HTTP BASIC authentication.
+// - password: (Required) ServiceNow password to use for HTTP BASIC authentication.
+// - description: (Required) Event description.
+// - severity: (Required) Severity of the event.
+//   Supported values:
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:23]:
```suggestion
//
//   Supported values:
//
```

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 2]
```diff
@@ -1,61 +1,77 @@
...
+//     node: lastReported.host,
+//     metricType: lastReported._measurement,
+//     resource: lastReported.instance,
+//     metricName: lastReported._field,
+//     severity:
+//         if lastReported._value < 1.0 then "critical"
+//         else if lastReported._value < 5.0 then "warning"
+//         else "info",
+//     additionalInfo: {"devId": r.dev_id}
+// )
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:73]: `flux fmt`'d

```suggestion
import "contrib/bonitoo-io/servicenow"
import "influxdata/influxdb/secrets"

username = secrets.get(key: "SERVICENOW_USERNAME")
password = secrets.get(key: "SERVICENOW_PASSWORD")

lastReported = from(bucket: "example-bucket")
    |> range(start: -1m)
    |> filter(fn: (r) => r._measurement == "cpu" and r._field == "usage_idle")
    |> last()
    |> findRecord(fn: (key) => true, idx: 0)

servicenow.event(
    url: "https://tenant.service-now.com/api/global/em/jsonv2",
    username: username,
    password: password,
    node: lastReported.host,
    metricType: lastReported._measurement,
    resource: lastReported.instance,
    metricName: lastReported._field,
    severity: if lastReported._value < 1.0 then
        "critical"
    else if lastReported._value < 5.0 then
        "warning"
    else
        "info",
    additionalInfo: {"devId": r.dev_id},
)
```

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 3]
```diff
@@ -1,61 +1,77 @@
...
 // ```
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:74]:
~~~suggestion
// ```
//
// tags: single notification
//
~~~

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 4]
```diff
@@ -102,53 +118,65 @@ event = (
...
-//        severity:
-//            if r._value < 1.0 then "critical"
-//            else if r._value < 5.0 then "warning"
-//            else "info",
-//        additionalInfo: { "devId": r.dev_id }
-//      })
-//    )()
+// - url: (Required) ServiceNow web service URL.
+// - username: (Required) ServiceNow username to use for HTTP BASIC authentication.
+// - password: (Required) ServiceNow password to use for HTTP BASIC authentication.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:126]: The "required" tags aren't needed. If a parameter is required, that metadata is included in the generated fluxdoc JSON and we can dynamically apply styles using that metadata.

```suggestion
// - url: ServiceNow web service URL.
// - username: ServiceNow username to use for HTTP BASIC authentication.
// - password: ServiceNow password to use for HTTP BASIC authentication.
```

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 5]
```diff
@@ -102,53 +118,65 @@ event = (
...
+//
+// ## Usage:
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:128]:
```suggestion
// ## Usage
```

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 6]
```diff
@@ -102,53 +118,65 @@ event = (
...
+// - `severity`
+// - `source`
+// - `node`
+// - `metricType`
+// - `resource`
+// - `metricName`
+// - `messageKey`
+// - `additionalInfo`
+//
+// (For more information, see `servicenow.event()` parameters.)
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:147]:
```suggestion
// For more information, see `servicenow.event()` parameters.
```

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 7]
```diff
@@ -102,53 +118,65 @@ event = (
...
+//     |> endpoint(mapFn: (r) => ({
+//         node: r.host,
+//         metricType: r._measurement,
+//         resource: r.instance,
+//         metricName: r._field,
+//         severity: "critical",
+//         additionalInfo: { "devId": r.dev_id }
+//       })
+//     )()
 // ```
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:179]:
```suggestion
// ```
//
// tags: notification endpoints
//
```

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 8]
```diff
@@ -1,61 +1,77 @@
...
+// ## Parameters:
+//
+// ServiceNow Event API fields are described in
+// [ServiceNow Create Event documentation](https://docs.servicenow.com/bundle/paris-it-operations-management/page/product/event-management/task/t_EMCreateEventManually.html).
+//
+// - url: (Required) ServiceNow web service URL.
+// - username: (Required) ServiceNow username to use for HTTP BASIC authentication.
+// - password: (Required) ServiceNow password to use for HTTP BASIC authentication.
+// - description: (Required) Event description.
+// - severity: (Required) Severity of the event.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:22]:
```suggestion
// - url: ServiceNow web service URL.
// - username: ServiceNow username to use for HTTP BASIC authentication.
// - password: ServiceNow password to use for HTTP BASIC authentication.
// - description: Event description.
// - severity: Severity of the event.
```

**Review by sanderson — CHANGES_REQUESTED** [review]

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 9]
```diff
@@ -102,53 +124,68 @@ event = (
...
+//         resource: r.instance,
+//         metricName: r._field,
+//         severity: "critical",
+//         additionalInfo: { "devId": r.dev_id }
+//       })
+//     )()
 // ```
 //
+// tags: notification endpoints
+//
```
Comment (nathanielc) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:188]: same

**Review by nathanielc — APPROVED** [review]
Looks good after a few small changes, no need for another review.

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux (hunk 9 above)
Comment (pierwill) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:188]:
```suggestion
// tags: notification endpoints
```

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 10]
```diff
@@ -1,61 +1,83 @@
...
+//     else if lastReported._value < 5.0 then
+//         "warning"
+//     else
+//         "info",
+//     additionalInfo: {"devId": r.dev_id},
+// )
 // ```
 //
+// tags: single notification
+//
```
Comment (pierwill) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:80]:
```suggestion
// tags: single notification
```

File: stdlib/contrib/bonitoo-io/servicenow/servicenow.flux [hunk 11]
```diff
@@ -1,61 +1,82 @@
-// Package servicenow provides functions for sending data to ServiceNow.
+// Package servicenow  provides functions for sending events to ServiceNow.
```
Comment (sanderson) [inline stdlib/contrib/bonitoo-io/servicenow/servicenow.flux:1]:
```suggestion
// Package servicenow  provides functions for sending events to [ServiceNow](https://www.servicenow.com/).
```

**Review by sanderson — APPROVED** [review]

**Pull request merged**